extensions/opentelemetry/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/tracing/TracerProcessor.java

```diff
@@ -151,8 +151,7 @@ void dropNames(Optional<StaticResourcesBuildItem> staticResources,
         if (config.isPropertyPresent("quarkus.http.root-path")) {
             String rootPath = config.getValue("quarkus.http.root-path", new NormalizeRootHttpPathConverter());
             String nonApplicationRootPath = config.getRawValue("quarkus.http.non-application-root-path");
-            // span names don't include the leading slash
-            nonApplicationUris.add(rootPath.substring(1) + nonApplicationRootPath);
+            nonApplicationUris.add(rootPath + nonApplicationRootPath);
```

Comment:
This is problematic, and is where you should be using NonApplicationPath build items instead. There are nuances here about when things are concatenated vs. not that aren't being taken into account.

This may mean moving http instrumentation into a processor (or method) that is only touched when Vertx is enabled, e.g. .. but directly reading config is not a good plan.

Reply:
We are already aware of this and I would prefer to handle this on a separate PR. Is that ok?